src/workerd/util/autogate.c++

```diff
@@ -78,4 +100,18 @@ void Autogate::initAutogateNamesForTest(std::initializer_list<kj::StringPtr> gat
   Autogate::initAutogate(gates.asReader());
 }
 
+bool Autogate::isAllAutogatesEnabled() {
+  initializeIfNot();
```

Comment:
@kentonv I'm extremely unhappy about this. If the code doesn't check for isAllAutogatesEnabled() or isEnabled(), we won't initialize all autogates. I think we need to revert and use the first solution I've did, and initialize it on a separate cpp binary.

Referencing: https://github.com/cloudflare/workerd/commit/982740115da30613698a79b5787fd96d3604726a#diff-a66344587988d22c97a0e51d030a064225c81d3f12efe46b989b6c5483e96b5bR18-R22